tsconfig.json

```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "lib": ["es2019", "dom"],
+    "lib": ["es2023", "dom"],
```

Comment:
This change makes me nervous, I'm not sure I know all the side effects. Is this needed for this PR? If not then I would suggest separating out to its own PR and explaining this change a bit.

Reply:
This change is needed for TypeScript compiler to recognize that `findLast` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLast) function exists on Array. Broader implications - making it recognize other library functions that were introduced in ES2023 standard